src/ray/raylet/node_manager.cc

```diff
@@ -519,20 +519,27 @@ void NodeManager::FillResourceReport(rpc::ResourcesData &resources_data) {
   cluster_resource_scheduler_->FillResourceUsage(resources_data);
   cluster_task_manager_->FillResourceUsage(resources_data);
 
+  // If plasma store is under high pressure, we should try to schedule a global gc.
+  bool plasma_high_pressure =
+      object_manager_.GetUsedMemoryPercentage() > high_plasma_storage_usage_;
```

Comment:
Should we adjust the interval based on the current memory usage instead of using a static threshold/interval?

Also, what are the guarantees for when this method gets called? Is it on a timer?

Reply:
Ah, I've talked with @ericl about this, the conclusion is that the 30s probably should be the minimum duration to avoid overhead to the cluster. The default GC on the local node is 10min here. And here it's to issue a global gc which is different from that one.

Besides, we think maybe the global GC should be sent from GCS since that's the place that can have better global control. For example, if all nodes are at a similar usage of memory, global GC will be triggered at the same time, and it's N*N message passing which is not necessary. Although it goes with the heartbeat, the heartbeat will be forced to be broadcasted if global_gc bit is set to be true.

This goes with report, it's run periodically ([code](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L421-L423))

Reply:
Ah I see, thanks for explaining!

Yeah for the report, I was just worried we might miss the GC call because this handler doesn't get called soon enough. This seems fine, though.

Reply:
I like the gcs based report idea. Can you create a enhancement issue? Would there be any issue with that approach?

Reply:
> Ah I see, thanks for explaining!
> 
> Yeah for the report, I was just worried we might miss the GC call because this handler doesn't get called soon enough. This seems fine, though.

I agree with you. I think here if memory usage is close to 100%, we probably can do nothing. It'll help if it bump to 90% suddenly and increase slowly. That's why even with this, it won't help fix the issue.